code/android-core-library/src/main/java/com/adobe/marketing/mobile/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,454 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.configuration
+
+import androidx.annotation.VisibleForTesting
+import com.adobe.marketing.mobile.Event
+import com.adobe.marketing.mobile.Extension
+import com.adobe.marketing.mobile.ExtensionApi
+import com.adobe.marketing.mobile.ExtensionEventListener
+import com.adobe.marketing.mobile.LoggingMode
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.internal.compatibility.CacheManager
+import com.adobe.marketing.mobile.launch.rulesengine.LaunchRulesEngine
+import com.adobe.marketing.mobile.launch.rulesengine.LaunchRulesEvaluator
+import com.adobe.marketing.mobile.services.CacheFileService
+import com.adobe.marketing.mobile.services.ServiceProvider
+import java.lang.Exception
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+
+/**
+ * Configuration Extension.
+ * Responsible for retrieving the configuration of the SDK, updating the shared state and
+ * dispatching configuration updates through the `EventHub`
+ */
+internal class ConfigurationExtension : Extension {
+
+    companion object {
+        private const val TAG = "ConfigurationExtension"
+        private const val EXTENSION_NAME = "com.adobe.module.configuration"
+        private const val EVENT_TYPE = "com.adobe.eventType.configuration"
+        private const val EVENT_SOURCE_REQUEST_CONTENT = "com.adobe.eventSource.requestContent"
+        private const val EVENT_SOURCE_RESPONSE_CONTENT = "com.adobe.eventSource.responseContent"
+
+        internal const val CONFIGURATION_REQUEST_CONTENT_JSON_APP_ID = "config.appId"
+        internal const val CONFIGURATION_REQUEST_CONTENT_JSON_FILE_PATH = "config.filePath"
+        internal const val CONFIGURATION_REQUEST_CONTENT_JSON_ASSET_FILE = "config.assetFile"
+        internal const val CONFIGURATION_REQUEST_CONTENT_UPDATE_CONFIG = "config.update"
```

Comment:
These constants are already used from our public APIs and are already present in CoreConstants file. Can you remove these and reuse them?